### DISCUSSION TITLE
Error: Could not find `source` context

### DISCUSSION THREAD

**Discussion by ehuelsda** [sanity-io/sanity#3654] [Sanity Studio Support]
I just started to familiarise myself with the v3 of the studio, using typescript. 
Unfortunately I'm running into issues trying to fetch data in my nextjs application, as the `useClient` hook gives me the following error: `Error: Could not find `source` context`. I'm afraid I'm missing something obvious here. Any pointers? 

The following code triggers the error: ` const client = useClient().withConfig({apiVersion: '2021-10-21'})`

I'm on 3.0.0-dev-preview.15

**Comment by Designer023** [comment]
I'm having the same issue. This has been present since the beta versions. I am currently on `3.5.1`

**Comment by DRITE** [comment]
I got the error while proceeding on `visionTool`.
Updating of `@sanity/vision` to the latest ([3.8.3](https://www.npmjs.com/package/@sanity/vision/v/3.8.3)) has fixed the problem for me.

**Comment by rselimi** [comment]
I'm on `3.11.1` and still see the same error

**Comment by fev4** [comment]
Same in `3.14.2`

I just migrated from v2 to sanity v3 and this is the only error stopping me from completing the migration.

Has anyone managed to fix this? How?

**Comment by fev4** [comment]
What managed to fix this for me was deleting `node_modules`, the lock file, maybe during upgrades something got funky and that was it. Additionally I removed the caret from the `sanity` package:
```
    "sanity": "^3.14.2", //before
    "sanity": "3.14.2", //after
    "@sanity/vision": "3.14.2",
```
Don't think that was it, but it was something suggested in other [threads](https://github.com/sanity-io/sanity/discussions/3792).

**Reply by rvande** [reply]
this worked for me, I was dealing with this at setup all day and ended up restarting, thank you for the help
